### PR TITLE
(hot fix) Resolve Boring App

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR resolves a bug introduced by this PR: https://github.com/Lightning-AI/lightning/pull/14464

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_app/utilities/app_logs.py [hunk 1]
```diff
@@ -86,18 +85,13 @@ def _app_logs_reader(
     for th in log_threads:
         th.start()
 
-    # Print logs from queue when log event is available
-    user_log_start = "<<< BEGIN USER_RUN_FLOW SECTION >>>"
```
Comment (tchaton) [inline src/lightning_app/utilities/app_logs.py:90]: This was too flaky and some lines were dropped.

**Review by awaelchli — APPROVED** [review]

File: src/lightning_app/testing/testing.py [hunk 2]
```diff
@@ -239,7 +239,7 @@ def run_app_in_cloud(app_folder: str, app_name: str = "app.py", extra_args: [str
     with tempfile.TemporaryDirectory() as tmpdir:
         env_copy = os.environ.copy()
         env_copy["PACKAGE_LIGHTNING"] = "1"
-        env_copy["LIGHTNING_DEBUG"] = "1"
+        env_copy["LIGHTNING_DEBUG"] = debug_mode
```
[the remark this answers is not in the document]
Comment (tchaton) [reply src/lightning_app/testing/testing.py]: Done !

**Review by ethanwharris — APPROVED** [review]

**Review by dmitsf — APPROVED** [review]
👍

**Review by hhsecond — APPROVED** [review]

File: src/lightning_app/testing/testing.py [hunk 3]
```diff
@@ -210,7 +208,9 @@ def run_cli(args) -> Generator:
 
 @requires("playwright")
 @contextmanager
-def run_app_in_cloud(app_folder: str, app_name: str = "app.py", extra_args: [str] = []) -> Generator:
+def run_app_in_cloud(
+    app_folder: str, app_name: str = "app.py", extra_args: List[str] = [], debug: bool = True
```
Comment (Borda) [inline src/lightning_app/testing/testing.py:212]:
```suggestion
    app_folder: str, app_name: str = "app.py", extra_args: Sequence[str] = tuple(), debug: bool = True
```
this is not imutable

**Pull request merged**